src/Raven.Client/Documents/Operations/OperationExecutor.DeleteByIndex.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Queries;
+
+namespace Raven.Client.Documents.Operations
+{
+    public partial class OperationExecutor
+    {
+        public Operation DeleteByIndex<T, TIndexCreator>(Expression<Func<T, bool>> expression, QueryOperationOptions options = null) where TIndexCreator : AbstractIndexCreationTask, new()
+        {
+            var indexCreator = new TIndexCreator();
+            return DeleteByIndex(indexCreator.IndexName, expression, options);
+        }
+
+        public Operation DeleteByIndex<T>(string indexName, Expression<Func<T, bool>> expression, QueryOperationOptions options = null)
+        {
+            IRavenQueryable<T> query;
+            using (var session = _store.OpenSession())
+            {
+                query = session.Query<T>(indexName).Where(expression);
+            }
```

Comment:
Make sure that the scope of the session is big enough to include the query generation